### ISSUE TITLE
Using the tidyverse for unit conversion in Vignette 2

### ISSUE THREAD

**Issue by kristenpeach** [NCEAS/metajam#110]
In Case or Vignette 2 the section on unit conversion uses primarily base R. The preceding sections in this vignette use primarily tidyverse commands. I will try to re-write the section on unit conversion to make it more compatible with the rest of the vignette.

**Comment by kristenpeach** [comment]
**Progress**
I added some annotation to multiple code chunks to try to improve clarity. In the fixing_units code chunk I used tidy language to replace missing data code ("-9999") with "NA" in two of the datasets. I suggest we move the Ammonium to Ammoniacal-nitrogen conversion (coeff_conv_NH4_to_NH4N <- 0.7764676534) from the 'Constants' code chunk to the 'fixing_units' code chunk. It took me a while to figure out where that equation came from and why I could not use udunits. If we want to be very careful that no one deletes it we can include it in both code chunks, although I know that duplication is generally bad practice.  In the 'fixing_units' code chunk I added a few lines using 'ud.are.convertible' to show users why we are using ud.convert for feet/meters and not NH4/NH4-N. 

**To-Do**
-Look for additional opportunities to 'tidy' in the vignette
-Go through the exercise again to make sure everything works and the code chunk annotation is sufficient
- Figure out what the first ~8 lines of the 'fixing_units' code chunk are actually doing...Why "Copy the units"?

**Comment by kristenpeach** [comment]
**Progress**
In the original 'fixing_units' code chunk two lines of base R are used to convert "foot" to "meters" in the data file and the associated attributes file, which is really nice. I was able to achieve the same thing using tidy commands but I convert the data file and the attributes file separately so it ends up being 4 lines of code. I am not sure which would be easier for the user to understand. But I included both. 

**To Do**
-Discuss changes with Julien and commit changes

**Comment by kristenpeach** [comment]
https://github.com/kristenpeach/metajam/blob/master/vignettes/dataset-batch-processing.Rmd

**Comment by kristenpeach** [comment]
**Progress**
I had a lot of trouble using mutate() to convert the NH4-N columns. I thought mutate() would replace values within an existing column rather than making a new column if you just used the same column name. 

In other disappointing news, I spent an unreasonable amount of time trying to use tidy commands to achieve what these lines from the old code achieved: luq_units_merged[which(luq_units_merged$attributeName == "Gage_Ht"), c("unit_RioIcacos", "unit_RioMameyesPuenteRoto")] <- "foot"

I tried so many different combinations of mutate/select/filter, replace_na(), case_when(), mutate_if(), str_replace_na(), replace() etc. I thought this would be incredibly simple but it somehow occupied several hours of my time today. 

To Do
-Look at the same problems with fresh eyes tomorrow
-Use the most basic tidy commands possible so this vignette is useful for future users

https://github.com/kristenpeach/metajam/blob/master/vignettes/dataset-batch-processing.Rmd

**Comment by mbjones** [comment]
@kristenpeach As an aside, mutate does indeed allow you to replace a column's values without making a new column.  Here's a toy example:

```r
> df <- data.frame(x=c(1,2,3), y=c(5,6,7))
> df %>% mutate(x=x*2.1)
    x y
1 2.1 5
2 4.2 6
3 6.3 7
>
```

**Comment by kristenpeach** [comment]
Thanks Matt and Julien! Matt that is a great example. The thing I was getting hung up on is how to replace only specific cells within a column. Julien came up with a nice combination of mutate() and ifelse() that does the trick. I am really happy I know how to do this now. Thank you both!

I used mutate() and ifelse() to change some of the data cleanup in the vignette. I replaced a few lines of base R code with na_if() to replace missing data codes with NAs. 

https://github.com/kristenpeach/metajam/blob/master/vignettes/dataset-batch-processing.Rmd
